implants/imix/src/run.rs

```diff
@@ -34,7 +36,7 @@ pub async fn handle_main() {
 }
 
 async fn run(cfg: Config) -> anyhow::Result<()> {
-    let mut agent = Agent::new(cfg)?;
+    let mut agent = Agent::new(cfg, GRPC::init()?)?;
```

Comment:
Is this where like SMB would go?

Reply:
yep! after we merge its literally just make SMB.rs that implements the Transport trait and then just swap out GRPC::init here with SMB::init and it should _just work_. atleast theoretically :)